cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2586,8 +2624,17 @@ public TopicsChangedListener getTopicsChangedListener() {
 
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        if (requestCode == locationPermissionRequestCode && (geofenceList == null || geofenceList.size() == 0)) {
-            this.initGeoFences();
+        switch (requestCode) {
+            case (pushNotificationPermissionRequestCode):
+                if ((grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) || this.ignoreDisabledNotificationPermission) {
+                    this.subscribe(true, pendingSubscribeCallbackListener);
+                }
```

Comment:
Add:

```
} else if (!this.ignoreDisabledNotificationPermission) {
  String error = "Can not subscribe because the notification permission has been denied by the user. You can call CleverPush.setIgnoreDisabledNotificationPermission(true) to still allow subscriptions, e.g. for silent pushes.";
  Log.d(LOG_TAG, error);
  
  if (subscribedCallbackListener != null) {
     subscribedCallbackListener.onFailure(new Exception(error));
  }
}
```

Reply:
okay added